credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java

```diff
@@ -16,20 +16,54 @@
 package io.syndesis.credential;
 
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.syndesis.dao.manager.DataManager;
+import io.syndesis.model.connection.Connector;
+
+import org.springframework.boot.autoconfigure.social.SocialProperties;
+import org.springframework.core.io.support.SpringFactoriesLoader;
+import org.springframework.util.ClassUtils;
 
 final class CredentialProviderRegistry implements CredentialProviderLocator {
 
-    private final Map<String, CredentialProvider> providers = new ConcurrentHashMap<>();
+    private final Map<String, CredentialProviderFactory> credentialProviderFactories;
 
-    @Override
-    public void addCredentialProvider(final CredentialProvider credentialProvider) {
-        providers.put(credentialProvider.id(), credentialProvider);
+    private final DataManager dataManager;
+
+    /* default */ static class ConnectorSettings extends SocialProperties {
+
+        public ConnectorSettings(final Connector connector) {
+            setAppId(requiredProperty(connector, Credentials.CLIENT_ID_TAG));
+            setAppSecret(requiredProperty(connector, Credentials.CLIENT_SECRET_TAG));
+        }
+
+        private static String requiredProperty(final Connector connector, final String tag) {
+            return connector.propertyTaggedWith(tag).orElseThrow(
+                () -> new IllegalArgumentException("No property tagged with `" + tag + "` on connector: " + connector));
+        }
+    }
+
+    public CredentialProviderRegistry(final DataManager dataManager) {
+        this.dataManager = dataManager;
+
+        credentialProviderFactories = SpringFactoriesLoader
+            .loadFactories(CredentialProviderFactory.class, ClassUtils.getDefaultClassLoader()).stream()
+            .collect(Collectors.toMap(CredentialProviderFactory::id, Function.identity()));
     }
 
     @Override
     public CredentialProvider providerWithId(final String providerId) {
-        final CredentialProvider providerWithId = providers.get(providerId);
+        final Connector connector = dataManager.fetch(Connector.class, providerId);
```

Comment:
This seems odd to fetch the `Connector` using `providerId`. Is this method really better called `providerForConnector` and `providerId` -> `connectorId`?

Reply:
Ah the `providerId === credentialsId` always?

Reply:
Yeah I've stumbled on that myself, provider comes from Spring Social and we use Connector to name the same thing in this context. I could do a PR to switch all usages of `provider` to `connector`, would that be OK?

Reply:
Perhaps just raise an issue for it now, it's not urgent, just slightly confusing naming.

Reply:
Will merge this as-is anyway - thanks!